src/cartesian/ErrorBar.tsx

```diff
@@ -7,20 +7,28 @@ import { Props as XAxisProps } from './XAxis';
 import { Props as YAxisProps } from './YAxis';
 import { D3Scale, DataKey } from '../util/types';
 import { filterProps } from '../util/ReactUtils';
+import { BarRectangleItem } from './Bar';
+import { LinePointItem } from './Line';
+import { ScatterPointItem } from './Scatter';
 
-interface ErrorBarDataItem {
+export interface ErrorBarDataItem {
   x: number;
   y: number;
   value: number;
   errorVal?: number[] | number;
 }
 
+export type ErrorBarDataPointFormatter = (
+  entry: BarRectangleItem | LinePointItem | ScatterPointItem,
+  dataKey: DataKey<any>,
+) => ErrorBarDataItem;
+
 interface InternalErrorBarProps {
   xAxis?: Omit<XAxisProps, 'scale'> & { scale: D3Scale<string | number> };
   yAxis?: Omit<YAxisProps, 'scale'> & { scale: D3Scale<string | number> };
   data?: any[];
   layout?: 'horizontal' | 'vertical';
-  dataPointFormatter?: (entry: any, dataKey: DataKey<any>) => ErrorBarDataItem;
```

Comment:
If we define a breaking change as one that can breaks a users TS project, removing `any` will always be breaking.

Reply:
yeah true that, if we get complaints we'll just have to say that its for the best of the project as we increase type safety